app/commands/git/sync.rb

```diff
@@ -0,0 +1,72 @@
+module Git
+  class Sync
+    include Mandate
+    initialize_with :track, :synced_to_git_sha
+
+    def call
+      sync! unless synced_to_head?
+    end
+
+    def sync!
+      raise NotImplementedError
+    end
+
+    def update_git_repo!
+      git_repo.update!
+    end
+
+    def filepath_in_diff?(filepath)
+      diff.each_delta.any? do |delta|
+        [delta.old_file[:path], delta.new_file[:path]].include?(filepath)
+      end
+    end
+
+    memoize
+    def git_repo
+      Git::Repository.new(track.slug, repo_url: track.repo_url)
+    end
+
+    memoize
+    def head_git_track
+      Git::Track.new(track.slug, git_repo.head_sha, repo: git_repo)
+    end
+
+    memoize
+    def synced_git_track
```

Comment:
I'm finding this name hard to reason about. Maybe:

```suggestion
    def current_git_track
```